packages/swap/src/utils/function.ts

```diff
@@ -31,3 +32,35 @@ export const handleExit = (cb: AnyFunction) => {
     exitHandlers.delete(cb);
   };
 };
+
+export function unreachable(value: never, message: string): never {
+  throw new Error(message);
+}
+
+export function calculateExpiryTime(args: {
+  chain: Chain;
+  startBlock?: bigint;
+  expiryBlock?: bigint | null;
+}) {
+  const { chain, startBlock, expiryBlock } = args;
+
+  if (startBlock == null || expiryBlock == null) {
+    return new Date();
+  }
+
+  let diff = Number(expiryBlock - startBlock);
```

Comment:
a little more descriptive:
```suggestion
  let blocksRemaining = Number(expiryBlock - startBlock);
```